src/app/service-offering/custom-service-offering/custom-service-offering.component.ts

```diff
@@ -1,24 +1,46 @@
-import { Component, Inject } from '@angular/core';
-import { FormControl, FormGroup, Validators } from '@angular/forms'
+import { Component, Inject, OnInit } from '@angular/core';
+import { FormControl, FormGroup, ValidatorFn, Validators } from '@angular/forms'
 import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material';
 
 import { ComputeOfferingViewModel } from '../../vm/view-models';
+import { Account } from '../../shared/models';
+
+function LimitValidator(cpuNumberLimit: number): ValidatorFn {
+  return function (control: FormControl) {
+    return control.value > cpuNumberLimit
+      ? { cpuLimitExceeded: true }
+      : null;
+  };
+}
 
 @Component({
   selector: 'cs-custom-service-offering',
   templateUrl: 'custom-service-offering.component.html',
   styleUrls: ['custom-service-offering.component.scss']
 })
-export class CustomServiceOfferingComponent {
+export class CustomServiceOfferingComponent implements OnInit {
   public offering: ComputeOfferingViewModel;
   public hardwareForm: FormGroup;
+  public account: Account;
+  public maxCpu: number;
+  public maxMemory: number;
 
   constructor(
     @Inject(MAT_DIALOG_DATA) data,
     public dialogRef: MatDialogRef<CustomServiceOfferingComponent>,
   ) {
-    const { offering } = data;
-    this.offering = offering;
+    this.offering = data.offering;
+    this.account = data.account;
+  }
+
+  public ngOnInit() {
```

Comment:
Why you use a component hook to do this instead if constructor?

Reply:
Mostly we use ngOnInit for all the initialization/declaration and avoid stuff to work in the constructor. The constructor should only be used to initialize class members but shouldn't do actual "work".